### ISSUE TITLE
`#%` macro doesn't detect dotted fields

### ISSUE THREAD

**Issue by scauligi** [hylang/hyrule#36]
The `#%` macro isn't detecting arguments if you access an argument's attributes using the sugared syntax:

```python
=> #% %1.attribute
lambda : hyx_Xpercent_signX1.attribute

=> #% (. %1 attribute)
lambda hyx_Xpercent_signX1: hyx_Xpercent_signX1.attribute
```

**Comment by Kodiologist** [comment]
I blame https://github.com/hylang/hy/issues/2096.
